src/statefulmodelinstance.cpp

```diff
@@ -27,15 +27,41 @@ using namespace InferenceEngine;
 
 namespace ovms {
 
-const Status extractSequenceId(const tensorflow::TensorProto& proto, uint64_t& sequenceId) {
+const Status StatefulModelInstance::extractSequenceId(const tensorflow::TensorProto& proto, uint64_t& sequenceId) {
+    if (!proto.tensor_shape().dim_size()) {
+        SPDLOG_DEBUG("[Model: {} version: {}] Sequence id tensor proto does not contain tensor shape information", getName(), getVersion());
+        return StatusCode::SPECIAL_INPUT_NO_TENSOR_SHAPE;
+    } else if (proto.tensor_shape().dim_size() != 1) {
+        SPDLOG_DEBUG("[Model: {} version: {}] Sequence id tensor proto shape has invalid number of dimensions. Expecting shape with one dimension", getName(), getVersion());
+        return Status(StatusCode::INVALID_NO_OF_SHAPE_DIMENSIONS, "Required shape for sequence_id is: (1)");
+    }
+
+    if (proto.tensor_shape().dim(0).size() != 1) {
+        SPDLOG_DEBUG("[Model: {} version: {}] Sequence id tensor proto shape has invalid shape. Expecting shape: (1)", getName(), getVersion());
+        return Status(StatusCode::INVALID_SHAPE, "Required shape for sequence_id is: (1)");
+    }
+
     if (proto.uint64_val_size() == 1) {
         sequenceId = proto.uint64_val(0);
         return StatusCode::OK;
     }
     return StatusCode::SEQUENCE_ID_BAD_TYPE;
 }
 
-const Status extractSequenceControlInput(const tensorflow::TensorProto& proto, uint32_t& sequenceControlInput) {
+const Status StatefulModelInstance::extractSequenceControlInput(const tensorflow::TensorProto& proto, uint32_t& sequenceControlInput) {
```

Comment:
These methods: `extractSequenceControlInput` and `extractSequenceId` are very similar and can be used for different purposes than handling control input/sequence id too. If the only difference is type (uint32 vs uint64) then maybe name them accordingly?

Reply:
Like `extractOneUint32` and `extractOneUint64`?

Reply:
I don't see any other usage of these methods than in sequence_id and sequence_control_input extraction in the near future. 
Can we stay with those very specific names to make the purpose of these methods obvious for now and when any other use case for these methods appears we would rename them?

Reply:
Seems fair